src/onchain_reading.js

```diff
@@ -4,14 +4,17 @@ const { decodeOrdersBN, decodeIndexedOrdersBN } = require("./encoding")
  * Returns an iterator yielding an item for each page of order in the orderbook that is currently being collected.
  * @param {BatchExchangeViewer} contract to query from
  * @param {number} pageSize the number of items to fetch per page
+ * @param {number} blockNumber the block number to execute the query at
  */
-const getOpenOrdersPaginated = async function* (contract, pageSize) {
+const getOpenOrdersPaginated = async function* (contract, pageSize, blockNumber) {
```

Comment:
What is the little * after function?

Reply:
Its a generator function, it allows it to `yield` values part-way through execution. Here is a simple example of it in action:
```js
function* foo() {
  yield 1
  yield 2
  yield 3
}

for (const x of bar()) {
  console.log(x)
}
// Outputs:
// 1
// 2
// 3
```

Reply:
This specific function uses it to `yield` each page of orders as it paginates through them.